CMakeLists.txt

```diff
@@ -16,97 +16,115 @@ option(BUILD_TESTING "Enable test (depends on googletest)." OFF)
 # it prominent in the GUI.
 option(BUILD_SHARED_LIBS "Build library as shared." OFF)
 
-#
-# library : cpu_features
-#
-
 include (CheckIncludeFile)
 include (CheckSymbolExists)
 
-check_include_file(dlfcn.h HAVE_DLFCN_H)
-check_symbol_exists(getauxval "sys/auxv.h" HAVE_STRONG_GETAUXVAL)
+macro(setup_include_and_definitions TARGET_NAME)
+  target_include_directories(${TARGET_NAME}
+    PUBLIC  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+    PRIVATE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include/internal>
+  )
+  target_compile_definitions(${TARGET_NAME}
+    PUBLIC STACK_LINE_READER_BUFFER_SIZE=1024
+  )
+endmacro()
 
-set(_HDRS
-  include/cpu_features_macros.h
-)
+set(PROCESSOR_IS_MIPS FALSE)
+set(PROCESSOR_IS_ARM FALSE)
+set(PROCESSOR_IS_AARCH64 FALSE)
+set(PROCESSOR_IS_X86 FALSE)
+set(PROCESSOR_IS_POWER FALSE)
+
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "^mips")
+  set(PROCESSOR_IS_MIPS TRUE)
+endif()
```

Comment:
why don't using `elif()` since the following code does not support multiple processor at the same time...